compiler/ir/ir.psi2ir/src/org/jetbrains/kotlin/psi2ir/generators/GeneratorExtensions.kt

```diff
@@ -32,4 +30,11 @@ open class GeneratorExtensions {
 
         companion object Instance : SamConversion()
     }
+
+    open fun computeFieldVisibility(descriptor: PropertyDescriptor): Visibility =
+        when {
```

Comment:
Suggestion to move this logic inside the `PropertyDescriptor.fieldVisibility` in `PropertyGenerator` and return null in this method, to have only truly platform-specific code in `GeneratorExtensions` implementations

Reply:
Done